### PR TITLE
Fixed sarif output issue

### LINKED ISSUE #5029: Included `fixes` in SARIF output
**Is your feature request related to a problem? Please describe.**
> it seems that the SARIF outputed by semgrep doesn't include the [fixes section](https://docs.oasis-open.org/sarif/sarif/v2.1.0/csprd01/sarif-v2.1.0-csprd01.html#_Ref530139075), making it impossible to use it to generate change suggestions

**Describe the solution you'd like**
--sarif should output the `fixes` field when an autofix is available

This will enable better integration to downstream tools like reviewdog: https://github.com/returntocorp/semgrep/pull/4870

### PR DESCRIPTION
**Pull request by ajbt200128**
PR checklist:

- [ ] Documentation is up-to-date
- [ ] Changelog is up-to-date
- [X] Change has no security implications (otherwise, ping security team)

### CODE REVIEW

File: semgrep/semgrep/formatter/sarif.py [hunk 1]
```diff
@@ -42,6 +44,42 @@ def _rule_match_to_sarif(rule_match: RuleMatch) -> Mapping[str, Any]:
...
+            "description": {"text": description_text},
+            "artifactChanges": [
+                {
+                    "artifactLocation": {"uri": str(rule_match.path)},
+                    "replacements": [
+                        {
+                            "deletedRegion": {
+                                "startLine": rule_match.start.line,
+                                "startColumn": rule_match.start.col,
+                                "endColumn": rule_match.end.col,
```
Comment (brendongo) [inline semgrep/semgrep/formatter/sarif.py:68]: What happens with multiline fixes?

File: semgrep/tests/e2e/snapshots/test_output/test_sarif_output_include_nosemgrep/results.sarif [hunk 2]
```diff
@@ -1,5 +1,6 @@
 {
   "$schema": "https://raw.githubusercontent.com/oasis-tcs/sarif-spec/master/Schemata/sarif-schema-2.1.0.json",
+  "fixes": [],
```
Comment (brendongo) [inline semgrep/tests/e2e/snapshots/test_output/test_sarif_output_include_nosemgrep/results.sarif:3]: Can you add a test that actually contains a fix object?

File: semgrep/semgrep/formatter/sarif.py [hunk 3]
```diff
@@ -42,6 +44,42 @@ def _rule_match_to_sarif(rule_match: RuleMatch) -> Mapping[str, Any]:
...
+    def _rule_match_to_sarif_fix(rule_match: RuleMatch) -> Optional[Mapping[str, Any]]:
+
+        # if rule_match.extra.get("dependency_matches"):
+        fixed_lines = rule_match.extra.get("fixed_lines")
+
+        description = rule_match.fix
+        if rule_match.fix_regex:
+            fix_regex = rule_match.fix_regex
+            description = (
+                f"s/{fix_regex.regex}/{fix_regex.replacement}/{fix_regex.count or 'g'}"
```
Comment (brendongo) [inline semgrep/semgrep/formatter/sarif.py]: count here is not actually how many that need to be replaced but is used internally for rule_match uniqueness when deduplicating rule_matches. Maybe just make the description generic for now? `Semgrep rule suggested fix`

File: semgrep/semgrep/formatter/sarif.py (hunk 1 above)
Comment (ajbt200128) [reply semgrep/semgrep/formatter/sarif.py:68]: my bad, missed the endLine field

File: semgrep/semgrep/formatter/sarif.py (hunk 3 above)
Comment (ajbt200128) [reply semgrep/semgrep/formatter/sarif.py]: Got it

File: semgrep/tests/e2e/snapshots/test_output/test_sarif_output_include_nosemgrep/results.sarif (hunk 2 above)
Comment (ajbt200128) [reply semgrep/tests/e2e/snapshots/test_output/test_sarif_output_include_nosemgrep/results.sarif:3]: Done in e8a3ad7db5c64827c160b7d83ca52aaf6a367211

**Review by brendongo — APPROVED** [review]
Nice! Mind just adding to the changelog?

**Comment by offensivedev** [comment]
Hey, the schema `https://raw.githubusercontent.com/oasis-tcs/sarif-spec/master/Schemata/sarif-schema-2.1.0.json` results in `404: Not Found`.  Might as well fix it in this PR?

**Comment by kristof-mattei** [comment]
@offensivedev https://raw.githubusercontent.com/oasis-tcs/sarif-spec/master/Documents/CommitteeSpecifications/2.1.0/sarif-schema-2.1.0.json schema found here!

**Pull request merged**